nat/traversal/pinger.go

```diff
@@ -280,16 +235,18 @@ func (p *Pinger) pingReceiver(conn *net.UDPConn, stop <-chan struct{}) error {
 		case <-stop:
 			return errNATPunchAttemptStopped
 		default:
+			// Add read deadline to prevent possible conn.Read hang when remote peer doesn't send ping ack.
+			conn.SetReadDeadline(time.Now().Add(p.pingConfig.Timeout * 2))
 			n, err := conn.Read(buf)
-			if err != nil {
+			// Set higher read deadline when NAT proxy is used.
+			conn.SetReadDeadline(time.Now().Add(12 * time.Hour))
```

Comment:
Not sure if this is correct to set read deadline of 12 hours here. My idea was to set it from consumer side only after whole ping is done and consumer still need to use proxy connection but now it will be executed for each ping.

Reply:
I was planning to change it later to reset this timeout in another PR to keep this as before.
But anyway resetting this read deadline now. No more 12 hours deadline.